web/src/components/hooks/useDownloadValues.test.jsx

```diff
@@ -0,0 +1,163 @@
+import { QueryClient, QueryClientProvider } from "react-query";
+import { act, renderHook } from "@testing-library/react-hooks";
+import { getValues, useDownloadValues } from "./useDownloadValues";
+
+describe("useDownloadValues", () => {
+  describe("GET", () => {
+    let queryClient;
+    let wrapper;
+    beforeEach(() => {
+      queryClient = new QueryClient();
+      wrapper = function wrapperFunc({ children }) {
+        return (
+          <QueryClientProvider client={queryClient}>
+            {children}
+          </QueryClientProvider>
+        );
+      };
+    });
+    it("calls _getValues", async () => {
+      const fetchValuesSpy = jest.fn(() => Promise.resolve());
+      const testConfig = {
+        _getValues: fetchValuesSpy,
+        _createObjectURL: jest.fn(() => "test"),
+        _revokeObjectURL: jest.fn(() => "test"),
+        appSlug: "test",
+        fileName: "test",
+      };
+
+      const { result, waitFor } = renderHook(
+        () => useDownloadValues(testConfig),
+        {
+          wrapper,
+        }
+      );
+
+      await act(async () => {
+        await result.current.download();
+      });
+      await waitFor(() => result.current.isSuccess);
+
+      const expectedFetchConfig = {
+        appSlug: testConfig.appSlug,
+      };
+
+      expect(result.current.variables).toEqual(undefined);
+      expect(fetchValuesSpy).toHaveBeenCalledTimes(1);
+      expect(fetchValuesSpy).toHaveBeenCalledWith(expectedFetchConfig);
+    });
+    it.todo(
+      "test the rest of this or figure out how to use react-query with it"
+    );
+  });
+  describe("getValues", () => {
+    it("calls fetch with the correct url and configuration", async () => {
+      const expectedBody = {
+        test: "test",
+      };
+      const blobSpy = jest.fn(() => Promise.resolve(expectedBody));
+      const _fetchValuesSpy = jest.fn(() =>
+        Promise.resolve({
+          ok: true,
+          blob: blobSpy,
+        })
+      );
+      const testAppSlug = "testAppSlug";
+      const testToken = "testToken";
+      const testAPIEndpoint = "testAPIEndpoint";
+      const testGetValuesConfig = {
+        _fetch: _fetchValuesSpy,
+        _token: testToken,
+        apiEndpoint: testAPIEndpoint,
+        appSlug: testAppSlug,
+      };
+
+      const expectedAPIEndpoint = `${testAPIEndpoint}/app/${testAppSlug}/values`;
+      const expectedResponse = {
+        data: expectedBody,
+      };
+      const expectedFetchConfig = {
+        method: "GET",
+        headers: {
+          Authorization: testToken,
+          "Content-Type": "application/blob",
+        },
+      };
+      await expect(getValues(testGetValuesConfig)).resolves.toEqual(
+        expectedResponse
+      );
+      expect(_fetchValuesSpy).toHaveBeenCalledTimes(1);
+      expect(_fetchValuesSpy).toHaveBeenCalledWith(
+        expectedAPIEndpoint,
+        expectedFetchConfig
+      );
+      expect(blobSpy).toHaveBeenCalledTimes(1);
+    });
```

Comment:
I like to break out mocks and expecteds into explicit variables so they're easier to read.